components/badge/index.css

```diff
@@ -50,13 +52,13 @@ governing permissions and limitations under the License.
 
   inline-size: auto;
 
-  padding-block: var(--spectrum-badge-neutral-texticon-text-padding-top) var(--spectrum-badge-neutral-texticon-text-padding-bottom);
```

Comment:
Our examples don't use an icon and so should ref `textonly`, that way left/right padding is the same.